src/lib/DatePickerComponent.android.js

```diff
@@ -153,7 +153,7 @@ export class DatePickerComponent extends React.Component {
               {this.props.iconLeft ? this.props.iconLeft : null}
               {placeholderComponent}
               <View style={this.props.valueContainerStyle}>
-                <TText tid="Value" style={this.props.valueStyle}>
+                <TText tid={`Value[${this.state?.date?.getTime() || "unknown"}]`} style={this.props.valueStyle}>
```

Comment:
Using a value as a test ID can cause problems:

1. You need to know what the value is before you can find the element.
2. The element may be there but just not have the expected value. It's better if the test finds the element and then checks the value.
3. The element may be there but be off screen at the time, hence not findable, but the test won't know why it can't find it.
4. There could be several elements with the same value.

You could get around this by splitting the test ID into path components like <Value>/theTestID and the search for "/theTestID" and then parse the value out of the found element's actual test ID.

There is however an argument that you should be testing the value as shown, rather than relying on an easily ingested value that is embedded in the test ID, otherwise the test isn't complete. Rather than parsing the date from the picker, consider building a string from the expected date and compare that string with the value shown in the picker.

Reply:
> Using a value as a test ID can cause problems:
> 
> 1. You need to know what the value is before you can find the element.
> 2. The element may be there but just not have the expected value. It's better if the test finds the element and then checks the value.
> 3. The element may be there but be off screen at the time, hence not findable, but the test won't know why it can't find it.
> 4. There could be several elements with the same value.
> 
> You could get around this by splitting the test ID into path components like `<Value>/theTestID` and then search for "/theTestID" and then parse the value out of the found element's actual test ID.

I've added a method in Locator which finds test IDs in this format (where the content inside the square braces can be dynamic and it'll still be found). It outputs similar error messages to the other Locator methods to help with debugging too.

> There is however an argument that you should be testing the value as shown, rather than relying on an easily ingested value that is embedded in the test ID, otherwise the test isn't complete. Rather than parsing the date from the picker, consider building a string from the expected date and compare that string with the value shown in the picker.

I see there being two types of test with dates / times, one where we're checking the value matches our expectation - for example testing formulas as default values and separately testing the text format works with specific examples (like what Casey has done in his [promotion check feature](https://github.com/axsy-dev/qa-tests/blob/adcb98772a35dcff0e174a70667289a2b096c225/AppTester/src/test/resources/features/promotion_check/kpi_validation.feature#L26)).